core/src/mmu.cpp

```diff
@@ -2,54 +2,51 @@
 
 #include "mmu.h"
 
-#include <utility>
+#include <algorithm>
 
 namespace n_e_s::core {
 
-uint8_t Mmu::read_byte(uint16_t addr) const {
-    return *translate_address(addr);
-}
+Mmu::Mmu() : mem_banks_() {}
 
-uint16_t Mmu::read_word(uint16_t addr) const {
-    return read_byte(addr) | read_byte(addr + 1) << 8;
+void Mmu::add_mem_bank(IMemBank *mem_bank) {
+    mem_banks_.push_back(mem_bank);
```

Comment:
It looks like the mmu takes ownership of these, so they need to either be unique ptrs or be deleted at some point.

Reply:
I vote for unique_ptrs :)